File: SmartBus.Website/Data/Entities/Trip.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SmartBus.Website.Data.Entities
{
    public enum TripStatus
    {
        Started,
        Finished,
        Canceled
    }
    public class Trip : IEntity
    {
        public int Id { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public int DriverId { get; set; }
        public Driver Driver { get; set; }
        public int BusId { get; set; }
        public Bus Bus { get; set; }
        public ICollection<Passage> Passages { get; set; }
        public Route Route { get; set; }
        public int RouteId { get; set; }
        public TripStatus TripStatus { get; set; }
    }
}

File: smart-bus-terminal/src/app/models/trip.ts
import { Driver } from './driver';
import { Route } from './route';
import { Bus } from './bus';

export enum TripStatus {
  started,
  finished,
  cancelled
}
export interface Trip {
  id: number;
  driver: Driver;
  route: Route;
  bus: Bus;
  routeId: number;
  driverId: number;
  busId: number;
  startDate: Date | string;
  nTripOfDay: number;
  tripStatus: number;
}
